src/App/App.js

```diff
@@ -2,39 +2,76 @@ import React, {Component} from 'react';
 import Header from '../Header/Header';
 import MovieContainer from '../MovieContainer/MovieContainer';
 import MovieDetails from '../MovieDetails/MovieDetails';
-import movieData from '../data.js';
+// import movieData from '../data.js';
 import './App.css';
+import { fetchSingleMovie, fetchMovies } from '../apiCalls';
 
 class App extends Component {
-  constructor() {
-    super();
-    this.state = {
-      movies: movieData.movies,
-      movie: {}
-    }
-  }
+	constructor() {
+		super();
+		this.state = {
+			movies: [],
+      movie: null,
+			error: '',
+			statusCode: 0,
+			isMovieDetails: false
+		}
+	}
 
-  getSingleMovie = (movieId) => {
-    const clickedMovie = {"movie": 
-    {id: 1, title: "Fake Movie Title", poster_path: "https://image.tmdb.org/t/p/original//7G2VvG1lU8q758uOqU6z2Ds0qpA.jpg", 
-    backdrop_path: "https://image.tmdb.org/t/p/original//oazPqs1z78LcIOFslbKtJLGlueo.jpg", 
-    release_date: "2019-12-04", overview: "Some overview that is full of buzzwords to attempt to entice you to watch this movie! Explosions! Drama! True love! Robots! A cute dog!", 
-    average_rating: 6, genres: [{id: 18, name:"Drama"}], budget:63000000, 
-    revenue:100853753, runtime:139, tagline: "It's a movie!" }}
+  componentDidMount = () => {
+		fetchMovies()
+		.then(data => {
+			if(typeof data === 'object') {
+				this.setState({ movies: data.movies })
+			} else {
+				this.setState({ statusCode: data })
+			}
+		})
+		.catch(error => this.setState({ error: error.message }))
+	}
+	
+	getSingleMovie = (id) => {
+		this.setState({ isMovieDetails: true })
+    fetchSingleMovie(id)
+    .then(data => {
+			if(typeof data === 'object') {
+				this.setState({ movie: data.movie })
+			} else {
+				this.setState({ statusCode: data })
+			}
+		})
+		.catch(error => this.setState({ error: error.message }))
```

Comment:
Good job handling the error. Is there some logic in the application that will also remove this error? One thing you want to be careful of is having _stale_ errors held in the state. Once the user has been informed of the error sufficiently and allowed to correct it, we want to get that message out of state. Otherwise, the user might see an error message in the future when they shouldn't.